common/keyboard.go

```diff
@@ -102,89 +173,82 @@ func (k *Keyboard) down(key string) error {
 func (k *Keyboard) insertText(text string) error {
 	action := input.InsertText(text)
 	if err := action.Do(cdp.WithExecutor(k.ctx, k.session)); err != nil {
-		return fmt.Errorf("unable to send character: %w", err)
+		return fmt.Errorf("cannot execute insert text: %w", err)
 	}
 	return nil
 }
 
 func (k *Keyboard) keyDefinitionFromKey(keyString keyboardlayout.KeyInput) keyboardlayout.KeyDefinition {
-	shift := k.modifiers & ModifierKeyShift
-	keyDef := keyboardlayout.KeyDefinition{}
 	layout := keyboardlayout.GetKeyboardLayout(k.layoutName)
-	code := keyString
 	srcKeyDef, ok := layout.Keys[keyString]
-
+	// Find based on key value instead of code
+	if !ok {
+		for key, def := range layout.Keys {
+			if def.Key != string(keyString) {
+				continue
+			}
+			keyString, srcKeyDef = key, def
+			ok = true // don't look for a shift key below
```

Comment:
I think this would be slightly more efficient, since it would break out of the loop as soon as a match is found:
```suggestion
			if def.Key == string(keyString) {
				keyString, srcKeyDef = key, def
				ok = true // don't look for a shift key below
				break
			}
```

Same for the `ShiftKey` check below.

Reply:
Please see: 21cc7b9 and then refactored f4014fe.